server/src/Core.hs

```diff
@@ -169,14 +171,15 @@ runCommandLoop core = go
 
 -- | Post an update to the core's update queue (read by the websocket subscribers)
 postUpdate :: Path -> Core -> IO ()
-postUpdate path core = atomically $ do
-  value <- Persistence.getValue (coreCurrentValue core)
-  full <- isFullTBQueue (coreUpdates core)
-  -- In order not to block the reader thread, and subsequently stop processing coreQueue,
-  -- we don't send new updates to subscribers if coreUpdates is full.
-  if full then
-    return ()
-    else writeTBQueue (coreUpdates core) (Just $ Updated path value)
+postUpdate path core = do
+  isWsQueueFull <- atomically $ do
+    value <- Persistence.getValue (coreCurrentValue core)
+    full <- isFullTBQueue (coreUpdates core)
+    -- In order not to block the reader thread, and subsequently stop processing coreQueue,
+    -- we don't send new updates to subscribers if coreUpdates is full.
+    when (not full) $ writeTBQueue (coreUpdates core) (Just $ Updated path value)
+    return full
+  when (not isWsQueueFull) $ for_ (coreMetrics core) Metrics.incrementWsQueueAdded
```

Comment:
Then here you would just decide which counter to increment based on whether it's full or not.